apps/console/src/features/console-settings/components/console-roles/console-roles-edit/console-role-permissions.tsx

```diff
@@ -0,0 +1,288 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+    AlertInterface,
+    AlertLevels,
+    IdentifiableComponentInterface,
+    LoadableComponentInterface,
+    RolePermissionInterface,
+    RolesInterface
+} from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import {
+    EmphasizedSegment,
+    Heading,
+    PrimaryButton
+} from "@wso2is/react-components";
+import isEmpty from "lodash-es/isEmpty";
+import React, { FunctionComponent, ReactElement, useMemo, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Dispatch } from "redux";
+import { updateRoleDetails } from "../../../../roles/api/roles";
+import { Schemas } from "../../../../roles/constants/role-constants";
+import { CreateRolePermissionInterface, PatchRoleDataInterface } from "../../../../roles/models/roles";
+import useGetAPIResourceCollections from "../../../api/use-get-api-resource-collections";
+import {
+    APIResourceCollectionInterface,
+    APIResourceCollectionPermissionCategoryInterface,
+    APIResourceCollectionPermissionScopeInterface
+} from "../../../models/console-roles";
+import { SelectedPermissionCategoryInterface, SelectedPermissionsInterface } from "../../../models/permissions-ui";
+import transformResourceCollectionToPermissions from "../../../utils/transform-resource-collection-to-permissions";
+import CreateConsoleRoleWizardPermissionsForm from
+    "../create-console-role-wizard/create-console-role-wizard-permissions-form";
+
+/**
+ * Props interface of {@link ConsoleRolePermissions}
+ */
+export interface ConsoleRolePermissionsProps extends LoadableComponentInterface, IdentifiableComponentInterface {
+    /**
+     * Is read only.
+     */
+    isReadOnly?: boolean;
+    /**
+     * All the details about the role.
+     */
+    role: RolesInterface;
+    /**
+     * Handle role update callback.
+     */
+    onRoleUpdate: (activeTabIndex: number) => void;
+    /**
+     * Tab index
+     */
+    tabIndex: number;
+}
+
+/**
+ * Component to render the tenant & organization permission trees.
+ *
+ * @param props - Props injected to the component.
+ * @returns ConsoleRolePermissions component.
+ */
+const ConsoleRolePermissions: FunctionComponent<ConsoleRolePermissionsProps> = (
+    props: ConsoleRolePermissionsProps
+): ReactElement => {
+    const {
+        isReadOnly,
+        role,
+        onRoleUpdate,
+        tabIndex,
+        [ "data-componentid" ]: componentId
+    } = props;
+
+    const dispatch: Dispatch = useDispatch();
+
+    const { t } = useTranslation();
+
+    const { data: tenantAPIResourceCollections } = useGetAPIResourceCollections(true, "type eq tenant", "apiResources");
+
+    const { data: organizationAPIResourceCollections } = useGetAPIResourceCollections(
+        true,
+        "type eq organization",
+        "apiResources"
+    );
```

Comment:
Shall we pass shouldFetch param to these API calls so that they wouldn't run in each render?